R/do_adjustRtime-functions.R

```diff
@@ -582,3 +582,283 @@ adjustRtimeSubset <- function(rtraw, rtadj, subset,
+    param@nChromPeaks <- vapply(cp_raw, nrow, numeric(1))
+    param@rtMap <- bplapply(cp_raw, FUN = function(x) {
+        .match_reference_anchors(obs_peaks = x, ref_anchors = param@lamas,
+                                 ppm = param@ppm, tolerance = param@tolerance,
+                                 toleranceRt = param@toleranceRt)},
+    BPPARAM = BPPARAM)
+    param
+}
+
+#' @title Summary of LamaParama retention time alignment
+#'
+#' @description
+#' The `summarizeLamaMatch()` generates a summary of the LamaParama method.
+#' Composed of coverage % of the chrompeaks match over the total chrompeaks of
+#' the object. as well as a summary of the model that will be applied to the
+#' file to adjust the retention times
+#'
+#' @param `LamaParama` same object that will be passed to the `adjustRtime()`
+#' function. To run this function the `matchLamasChromPeaks()` need to be run
+#' on this first.
+#'
+#' @return A `data.frame` with:
+#'
+#' - Total_peaks: total number of chromatographic peaks in the file
+#' - Matched_peak: The number of matched peaks to Lamas
+#' - Total_Lamas: Total number of Lamas
+#' - Model_summary: `summary.loess` or `summary.gam` object for each file.
+#'
+#' @author Philippine Louail, Carl Brunius
+#'
+#' @rdname adjustRtime
+summarizeLamaMatch <- function(param){
+    if (!inherits(param, "LamaParama"))
+        stop("The input needs to be of class 'LamaParama'")
+    if (length(param@nChromPeaks) == 0 || length(param@rtMap) == 0)
+        stop("Summary inputs are missing. Please run `matchLamasChromPeaks` ",
+             "first.")
+    res <- data.frame(Total_peaks = param@nChromPeaks,
+                      Matched_peaks = vapply(param@rtMap, nrow, numeric(1)),
+                      Total_lamas = nrow(param@lamas))
+    res_model <- lapply(param@rtMap, function(x){
+         s <- summary(.rt_model(method = param@method,
+                      rt_map= x, span = param@span,
+                      resid_ratio = param@outlierTolerance,
+                      zero_weight = param@zeroWeight,
+                      bs = param@bs))
+        })
+    res$Model_summary <- res_model
+    res
+}
+
+#' @title Perform linear interpolation for unsorted retention time.
+#'
+#' @description
+#' This function performs linear interpolation on the non-sorted parts of an
+#' input vector of retention time. To see more details on the interpolation,
+#' see [approx()]
+#'
+#' @param rtime `numeric` vector with the retention times for one file/sample.
+#'
+#' @return vector with sorted retention time.
+#'
+#' @examples
+#' x <- c(NA, NA, NA, 1.2, 1.1, 1.14, 1.2, 1.3, 1.1, 1.04, 1.4, 1.6, NA, NA)
+#' sorted_rtime <-  .force_sorted(x)
+#' is.unsorted(x, na.rm = TRUE)
+#'
+#' @noRd
+.force_sorted <- function(x){
+    # Select only the non-NA values
+    nna_idx <- which(!is.na(x))
+    vec_temp <- x[nna_idx]
+
+    while (any(diff(vec_temp) < 0)) {
+        # had to change because which.max gives 1 if the vector is all FALSE..
```

Comment:
ah, that's interesting.

Reply:
an alternative would also be `idx <- match(TRUE, diff(vec_temp) < 0)` , but that is slightly slower.

Reply:
Yes I thought about `match()` but it gives NA when the vector is ordered so i can't have it as the `while` condition. and outside of the condition statement i think `which.max` is better ?

Reply:
yes, I think your approach is better.